src/Tools/Common/Commands/Utils.cs

```diff
@@ -42,15 +42,35 @@ public static int FindProcessIdWithName(string name)
         }
 
         /// <summary>
-        /// A helper method for validating --process-id, --name, --diagnostic-port options for collect and monitor commands.
+        /// A helper method for validating --process-id, --name, --diagnostic-port options for collect with child process commands.
+        /// None of these options can be specified, so it checks for them and prints the appropriate error message.
+        /// </summary>
+        /// <param name="processId">process ID</param>
+        /// <param name="name">name</param>
+        /// <param name="port">port</param>
+        /// <returns></returns>
+        public static bool ValidateArgumentsForChildProcess(int processId, string name, string port)
```

Comment:
Definitely not something I expect to be done in this PR, but I think we should start looking at wrapping this type of validation into a `{T}Validator` in System.CommandLine.  It shouldn't be too much work since most of these types of validations are already pulled out into util files/classes.